src/shared/components/ellipsisTextTooltip/EllipsisTextTooltip.tsx

```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
+import {action, observable} from 'mobx';
+import {observer} from "mobx-react";
+import styles from "./EllipsisTextTooltip.module.scss";
+import DefaultTooltip from "../defaultTooltip/DefaultTooltip";
+import $ from "jquery";
+import autobind from "autobind-decorator";
+
+@observer
+export default class EllipsisTextTooltip extends React.Component<{ text:string },{}> {
+
+    @observable tooltipVisible = false;
+
+    el:HTMLSpanElement;
+
+    @autobind
+    @action
+    onVisibleChange(isVisible:boolean){
+        const isOverflowed = (this.el.scrollWidth - $(this.el).innerWidth()) > 1;
```

Comment:
I feel like isOverflowed should be updated during componentDidMount, isVisible should be updated by onVisibleChange, and then tooltipVisible should be a computed that combines isVisible and isOverflowed